File: server/match_start_test.go
package main

import "testing"

func TestPlayableMatchRoster(t *testing.T) {
	s := newMatch()
	if s.RestartPhase != 1 || s.Time != 90 {
		t.Fatal("kickoff")
	}
	for _, p := range s.Players {
		for j, v := range p.Stats {
			if v != 170 || p.BaseStats[j] != 170 {
				t.Fatal("roster attributes")
			}
		}
		if movementSpeed(&p, false, false) != 6*velocityUnit || movementSpeed(&p, true, false) != 5*velocityUnit {
			t.Fatal("match pace")
		}
	}
	for _, b := range s.Bench {
		for _, stats := range b {
			for _, v := range stats {
				if v != 170 {
					t.Fatal("reserve attributes")
				}
			}
		}
	}
	s.Players[7].Stats = defaultStats()
	s.pickup(7, 11)
	for _, v := range s.Players[7].Stats {
		if v != 170 {
			t.Fatal("full energy baseline")
		}
	}
}

File: src/match-start.test.js
import test from "node:test";
import assert from "node:assert/strict";
import { createMatch } from "./game.js";
import { movementSpeed,velocityUnit } from "./attributes.js";
import { pickup } from "./features.js";
test("playable match starts use original two-player attributes for players and reserves", () => {
 for(const training of [false,true]) {
  const s=createMatch({training});assert.equal(s.restartPhase,1);assert.equal(s.time,90);
  for(const p of s.players) {
   assert.deepEqual(p.stats,Array(8).fill(170));assert.deepEqual(p.baseStats,Array(8).fill(170));
   assert.equal(movementSpeed(p,false),6*velocityUnit);assert.equal(movementSpeed(p,true),5*velocityUnit);
  }
  for(const b of s.bench) for(const stats of b) assert.deepEqual(stats,Array(8).fill(170));
  s.players[7].stats.fill(100);pickup(s,7,11);assert.deepEqual(s.players[7].stats,Array(8).fill(170));
  assert.equal(s.training,training);
 }
});
